src/HoC.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {ensure} from './util';
+import {update, unmount} from './Mixin';
+
+// Mixin for RethinkDB query subscription support in React components. You'll
+// generally want to use DefaultHoC or PropsHoC, which use BaseHoC to
+// create more usable versions.
+//
+// In your component, you should define an observe(props, state) function that
+// returns an object mapping query names to QueryRequests. See
+// QueryRequest.js for the API.
+//
+// In the child component, you will have access to this.props.data, which is an
+// object mapping from the same query names returned in observe() to the
+// results of each query as an QueryResult. See QueryResult.js for the
+// API.
+//
+// Here is a simple example of the mixin API:
+// const observe = (props, state) => ({
+//     turtles: new QueryRequest({
+//       query: r.table('turtles'),
+//       changes: true,
+//       initial: [],
+//     }),
+//   });
+
+// class App extends Component {
+//   render() {
+//     return <div>
+//       {this.props.data.turtles.value().map(function(x) {
+//         return <div key={x.id}>{x.firstName}</div>;
+//       })};
+//     </div>;
+//   },
+// };
+
+// BaseHoC(new Session())(observe)(App);
```

Comment:
The example shows `new Session()` being passed as `sessionGetter`. Should that be `() => new Session()`?

I'd also be fine not including an example for `BaseHoC` since `DefaultHoC` and `PropsHoC` are probably more useful for most developers.